src/terminal/parser/stateMachine.cpp

```diff
@@ -1848,7 +1848,7 @@ void StateMachine::ProcessString(const std::wstring_view string)
         if (_processingIndividually)
         {
             // If we're processing characters individually, send it to the state machine.
-            ProcessCharacter(string.at(current));
+            ProcessCharacter(til::at(string, current));
             ++current;

```

Comment:
i'd rather we keep these changes in the other PR that touches state machine -- mind backing them out from this one?

Reply:
These actually belongs to @lhecker. Sure, I can steal them way 😃